spec/delivery_spec.rb

```diff
@@ -1,95 +1,154 @@
 require_relative "spec_helper"
 
 describe CourierRails::DeliveryMethod do
-  before(:each) do
-    @delivery_method = CourierRails::DeliveryMethod.new
-  end
+  # before(:each) do
+  #   @delivery_method = CourierRails::DeliveryMethod.new
+  # end
+  let(:delivery_method) { CourierRails::DeliveryMethod.new }
 
   context "Event" do
     it "accepts the event to use" do
       test_email = Mailer.test_email courier_data: {event: "TEST_EVENT"}
 
-      @delivery_method.deliver!(test_email)
+      delivery_method.deliver!(test_email)
 
-      expect(@delivery_method.payload["event"]).to eq("TEST_EVENT")
+      expect(delivery_method.payload["event"]).to eq("TEST_EVENT")
     end
 
     it "raises exception if event is missing" do
       test_email = Mailer.test_email
 
-      expect { @delivery_method.deliver!(test_email) }.to raise_exception(StandardError)
+      expect { delivery_method.deliver!(test_email) }.to raise_exception(StandardError)
```

Comment:
It would be ideal to expect a specific exception here. I am not sure what specific error class would normally be raised here, but this test could fib to you.

`raise_exception(StandardError)` is functionally equivalent to `raise_exception(Exception)` and both of these will catch *any* exception. So you could be raising something unrelated that is broken elsewhere, and this test will still pass even though it isn't raising the error you expected, which could be a bit confusing if a bug actually comes out.

Not essential to patch up, but normally I stray away from testing for `StandardError` or `Exception` with expectations of `rase_error` or in a `rescue`.

Reply:
Yeah, this makes sense. Since this existed before this PR, I'm going to address it in a different PR as to not hold this one up. There's likely a lot of clean up that will need to be done.